### PR TITLE
coq: improve directory target message

### PR DESCRIPTION
**Pull request by Alizter**
Since this is in gen_rules @rgrinberg could you take a look.

### CODE REVIEW

File: src/dune_rules/gen_rules.ml [hunk 1]
```diff
@@ -305,10 +305,10 @@ let collect_directory_targets ~init ~dir =
           |> Path.Build.Map.union acc ~f:(fun path loc1 loc2 ->
                  User_error.raise
```
Comment (rgrinberg) [inline src/dune_rules/gen_rules.ml:306]: You could add `~loc:loc1` or `~loc:loc2` here

**Review by rgrinberg — APPROVED** [review]

**Comment by Alizter** [comment]
Hmm actually I'm not sure including the location there made it easier. I will revert.

**Comment by rgrinberg** [comment]
You mean the `~loc` arg? Without it, the editor won't be able to jump to the error

**Comment by Alizter** [comment]
Oh yes, you are correct. I've update the test that was affected and cleared up the wording (since anybody can create this dir target and not just a theory stanza).

**Comment by Alizter** [comment]
Do you think it matters to choose `loc1` or `loc2`. I'm thinking the latter might be more correct since it was found "later"?

**Comment by rgrinberg** [comment]
I don't think it matters much. In RPC, you can provide both locations anyway

**Pull request merged**